src/Middleware/Diagnostics.EntityFrameworkCore/src/DatabaseDeveloperPageExceptionFilter.cs

```diff
@@ -85,8 +85,10 @@ public async Task HandleExceptionAsync(ErrorContext errorContext, Func<ErrorCont
             catch (Exception e)
             {
                 _logger.DatabaseErrorPageMiddlewareException(e);
-                return;
             }
+
+            // Error could not be handled, let the next filter try
+            await next(errorContext);
```

Comment:
When the exception occurred matters. The following are checked before calling the handers at all:
https://github.com/dotnet/aspnetcore/blob/52eff90fbcfca39b7eb58baad597df6a99a542b0/src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddleware.cs#L100-L109
If one handler is going to fall back to another, it should do similar steps before calling next.

Reply:
Hmm now I think about it shouldn't we check for Response.HasStarted at the beginning of the filter as well?

Reply:
It's checked before the filter is called. If we follow the middleware pattern, it's each filters responsibility to only call next if the request is still viable.

Reply:
I suspect this is something that's easily missed, maybe we should be enforcing this somehow.